client.go

```diff
@@ -285,6 +278,66 @@ func (c *TunaSessionClient) shouldAcceptAddr(addr string) bool {
 	return false
 }
 
+func (c *TunaSessionClient) getPubAddrsFromRemote(ctx context.Context, remoteAddr string, config *nkn.DialConfig) (*PubAddrs, error) {
+	buf, err := json.Marshal(&Request{Action: "getPubAddr"})
+	if err != nil {
+		return nil, err
+	}
+
+	msgChan := make(chan *nkn.Message, 1)
+	errChan := make(chan error, 1)
+	doneCtx, doneCancel := context.WithCancel(ctx)
+	defer doneCancel()
+	wg := sync.WaitGroup{}
+	for i := 0; i < 3; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			select {
+			case <-time.After(time.Duration(3*i) * time.Second):
+			case <-doneCtx.Done():
+				return
+			}
+			respChan, err := c.multiClient.Send(nkn.NewStringArray(remoteAddr), buf, nil)
+			if err != nil {
+				select {
+				case errChan <- err:
+				default:
+				}
+				return
+			}
+			select {
+			case msg := <-respChan.C:
+				select {
+				case msgChan <- msg:
+				default:
+				}
+				doneCancel()
+			case <-doneCtx.Done():
+			}
+		}(i)
+	}
+	wg.Wait()
+	close(msgChan)
+	close(errChan)
+
+	msg, ok := <-msgChan
+	if !ok {
+		err, ok := <-errChan
+		if ok {
+			return nil, err
+		}
+		return nil, ctx.Err()
+	}
+
+	pubAddrs := &PubAddrs{}
+	err = json.Unmarshal(msg.Data, pubAddrs)
+	if err != nil {
+		return nil, err
+	}
+	return pubAddrs, nil
```

Comment:
Now with #46 being merged, each element in pubaddrs might be nil. We should check here and return error if all items in pubaddres are nil